includes/class-wc-calypso-bridge-filters.php

```diff
@@ -40,8 +41,43 @@ private function __construct() {
 
 		// Turn off email notifications.
 		add_filter( 'pre_option_woocommerce_merchant_email_notifications', array( $this, 'disable_email_notes' ) );
+
+		// Hides all WooCommerce Database update notices.
+		add_filter( 'woocommerce_rest_notes_object_query', array( $this, 'remove_update_note_type_from_args' ), 10, 2 );
+		// Run after Automattic\WooCommerce\Admin\Loader.
+		add_filter( 'woocommerce_components_settings', array( $this, 'component_settings' ), 30 );
+		add_filter( 'woocommerce_shared_settings', array( $this, 'component_settings' ), 30 );
 	}
 
+	/**
+	 * Adjust alert count to the component settings, to prevent loading state from being shown on db updates.
+	 *
+	 * @param array $settings Component settings.
+	 */
+	public function component_settings( $settings ) {
+		$settings['alertCount'] = Notes::get_notes_count( array( 'error' ), array( 'unactioned' ) );
+		return $settings;
+	}
+
+	/**
+	 * Remove the "update" note type from REST API requests to surpress db notices from being displayed.
+	 *
+	 * @param array  $args REST request args
+	 * @param object $request WP_REST_Request
+	 */
+	public function remove_update_note_type_from_args( $args, $request ) {
```

Comment:
I tried a few different approaches here, but this one seemed the most targeted to the use case of store alert notices being requested from the REST API.